tuf/download.py

```diff
@@ -257,14 +231,57 @@ def _download_file(url, required_length, STRICT_REQUIRED_LENGTH=True):
   temp_file = securesystemslib.util.TempFile()
 
   try:
-    # Open the connection to the remote file.  _open_connection() can raise
-    # socket connection exceptions (such as SSLError).  Connection errors of
-    # this kind can be minimized by adjusting the socket timeout in
-    # settings.py.
-    connection = _open_connection(url)
+    # Use a different requests.Session per hostname to reuse connections while
+    # minimizing subtle security issues.
+    hostname = six.moves.urllib.parse.urlparse(url).hostname
```

Comment:
Calls to fetch URLs using different protocols (http vs https) and different ports will share the same session. I don't know if that will result in some subtle security issue.

Reply:
That's a good point. Some cookies are [supposed](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies) to be over HTTPS only. Feel free to index sessions by "{scheme}+{hostname}" in there if you wish. That would basically be `netloc` IIRC.

Reply:
Resolved that way, with another commit